packages/core/src/baseCogniteClient.ts

```diff
@@ -107,16 +75,21 @@ export default class BaseCogniteClient {
    * ```
    */
   constructor(options: ClientOptions, apiVersion: CogniteAPIVersion = 'v1') {
-    verifyOptionsRequiredFields(options);
-
-    if (
-      options &&
-      !options.authentication?.credentials &&
-      !options.getToken &&
-      !options.noAuthMode
-    ) {
+    if (!options) {
+      throw Error('`CogniteClient` is missing parameter `options`');
+    }
+
+    if (!isString(options.appId)) {
+      throw Error('options.appId is required and must be of type string');
+    }
+
+    if (!isString(options.project)) {
+      throw Error('options.project is required and must be of type string');
+    }
```

Comment:
Feels like we can still use `verifyOptionsRequiredFields` in some form to manage this easier - but i'm also not sure how many new "required" things we would be adding to `options`.

Reply:
The `verifyOptionsRequiredFields` did a bit more and mostly related to auth options.
Keeping this as required for now as it's aligned with version 9.